### PR TITLE
[feat]: Drag and Drop Handles for all Data Structures

### PR DESCRIPTION
**Pull request by Palanikannan1437**
## Description

This PR introduces drag and drop support for ALL supported nodes in the editor, yup ALL of them!

- Tables
- Images
- Text/Paragraph
- Quotes
- Task List Item
- Ordered and Unordered Lists
- Code Blocks
- Headings

## Demo
https://github.com/makeplane/plane/assets/73993394/f04e033f-be3f-4945-9f22-2d1aaa652629

## Additional Fixes

1. Revamped Code Blocks
 	- Added Code Support with Syntax Highlighting by just typing in ```ts command.  
 	- Defaults to Plain text.
 	- You can directly copy paste from VSCode with Syntax Highlighting with no effort at all!!
 
       https://github.com/makeplane/plane/assets/73993394/b1011c0d-9c31-48da-9b83-99f34d62d4ad
 	- Tab keys work properly as intended
 	- Triple Enter doesn't exit out of code blocks anymore. You can exit the code block with CMD+Enter or Arrow Down/Up
 
2. Custom List keymaps added that fixes weird behaviour with List items and Task List items on hitting Backspace/Delete Key.
3. CMD/Ctrl+A selects the current block first and then the whole editor.

### CODE REVIEW

**Review by sriramveeraghanta — APPROVED** [review]

File: web/services/file.service.ts [hunk 1]
```diff
@@ -61,6 +63,14 @@ export class FileService extends APIService {
     this.cancelSource.cancel("Upload cancelled");
   }
 
+  // async validateFile(assetUrlWithWorkspaceId: string): Promise<any> {
```
Comment (rahulramesha) [inline web/services/file.service.ts:66]: please remove unused comments

File: packages/editor/rich-text-editor/src/ui/menus/bubble-menu/index.tsx [hunk 2]
```diff
@@ -125,6 +113,7 @@ export const EditorBubbleMenu: FC<EditorBubbleMenuProps> = (props: any) => {
             editor={props.editor!!}
             isOpen={isLinkSelectorOpen}
             setIsOpen={() => {
+              console.log("setIsLinkSelectorOpen");
```
Comment (rahulramesha) [inline packages/editor/rich-text-editor/src/ui/menus/bubble-menu/index.tsx:116]: remove debug console logs

File: packages/editor/rich-text-editor/src/ui/extensions/drag-drop.tsx [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+  function showDragHandle() {
+    if (dragHandleElement) {
+      dragHandleElement.classList.remove("hidden");
+    }
+  }
+
+  return new Plugin({
+    key: new PluginKey("dragHandle"),
+    view: (view) => {
+      dragHandleElement = document.createElement("div");
```
Comment (rahulramesha) [inline packages/editor/rich-text-editor/src/ui/extensions/drag-drop.tsx:138]: could you move the drag handle icon creation to a util

File: web/services/file.service.ts (hunk 1 above)
Comment (Palanikannan1437) [reply web/services/file.service.ts:66]: Done in https://github.com/makeplane/plane/pull/2789

File: packages/editor/rich-text-editor/src/ui/extensions/drag-drop.tsx (hunk 3 above)
Comment (Palanikannan1437) [reply packages/editor/rich-text-editor/src/ui/extensions/drag-drop.tsx:138]: done in https://github.com/makeplane/plane/pull/2789/commits/c5feaabc97e85a94feca9a56cc92558cfe259e20

File: packages/editor/rich-text-editor/src/ui/menus/bubble-menu/index.tsx (hunk 2 above)
Comment (Palanikannan1437) [reply packages/editor/rich-text-editor/src/ui/menus/bubble-menu/index.tsx:116]: done in https://github.com/makeplane/plane/pull/2789/commits/7085e6e2f6692005aede682344d1d6d423535713

**Pull request merged**